src/main/java/org/sonarlint/intellij/analysis/JavaAnalysisConfigurator.java

```diff
@@ -50,118 +47,42 @@
-    console.debug("Done in " + (System.currentTimeMillis() - start) + "ms\n");
-    return result;
-  }
-
-  private static Charset getEncoding(Project p, @Nullable VirtualFile f) {
-    if (f != null) {
-      Charset encoding = EncodingProjectManager.getInstance(p).getEncoding(f, true);
-      if (encoding != null) {
-        return encoding;
-      }
-    }
-    return Charset.defaultCharset();
-  }
-
-  private static List<ClientInputFile> getInputFiles(Project p, ModuleRootManager moduleRootManager, Collection<VirtualFile> filesToAnalyze) {
-    Collection<String> testFolderPrefix = findTestFolderPrefixes(moduleRootManager);
-    List<ClientInputFile> inputFiles = new LinkedList<>();
-
-    for (VirtualFile f : filesToAnalyze) {
-      boolean test = isTestFile(testFolderPrefix, f);
-      Charset charset = getEncoding(p, f);
-      inputFiles.add(new DefaultInputFile(f, test, charset));
-    }
-
-    return inputFiles;
-  }
-
-  private static boolean isTestFile(Collection<String> testFolderPrefix, VirtualFile f) {
-    String filePath = f.getPath();
-    for (String testPrefix : testFolderPrefix) {
-      if (filePath.startsWith(testPrefix)) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  private static void configureModuleSettings(@NotNull Module ijModule, @NotNull Map<String, String> properties) {
+  public Map<String, String> configure(@NotNull Module ijModule) {
+    Map<String, String> properties = new HashMap<>();
     configureLibraries(ijModule, properties);
     configureBinaries(ijModule, properties);
     configureJavaSourceTarget(ijModule, properties);
+    return properties;
   }
 
   private static void configureJavaSourceTarget(final Module ijModule, Map<String, String> properties) {
     try {
-      final String languageLevel = getLanguageLevelOption(ApplicationManager.getApplication()
-        .runReadAction((Computable<LanguageLevel>) () -> EffectiveLanguageLevelUtil.getEffectiveLanguageLevel(ijModule)));
+      LanguageLevel languageLevel = ApplicationManager.getApplication()
+        .runReadAction((Computable<LanguageLevel>) () -> EffectiveLanguageLevelUtil.getEffectiveLanguageLevel(ijModule));
+      final String languageLevelStr = getLanguageLevelOption(languageLevel);
       String bytecodeTarget = CompilerConfiguration.getInstance(ijModule.getProject()).getBytecodeTargetLevel(ijModule);
       if (StringUtil.isEmpty(bytecodeTarget)) {
         // according to IDEA rule: if not specified explicitly, set target to be the same as source language level
-        bytecodeTarget = languageLevel;
+        bytecodeTarget = languageLevelStr;
       }
-      properties.put(JAVA_SOURCE_PROPERTY, languageLevel);
+      properties.put(JAVA_SOURCE_PROPERTY, languageLevelStr);
       properties.put(JAVA_TARGET_PROPERTY, bytecodeTarget);
-    } catch (NoClassDefFoundError e) {
-      // CompilerConfiguration not available for example in PHP Storm
+    } catch (BootstrapMethodError | NoClassDefFoundError e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)